app/src/Containers/Logout/index.tsx

```diff
@@ -0,0 +1,15 @@
+import { Logout } from "./styles"
+import React from "react"
+import { useDispatch } from "react-redux"
+import { logoutUser } from "../../actions"
+
+const LoginOut = () => {
+  const dispatch = useDispatch()
+  return(
+    <>
+      <Logout onClick={() => dispatch(logoutUser())}>Log Out</Logout>
```

Comment:
this is a proper dispatch, let's go through what happens

```mermaid
graph TD
    A[reqLogout] -->|returns a thunk| B(dispatch calls that returned thunk, passing both itself and the global state to it)
    B --> C[gets a promise returned from the invoked thunk]
    C -->|waits for the promise to resolve| D[calls out to the backend to clear the session]
    D --> E[sends an object to the dispatch function the thunk was invoked with]
    E --> F[all of the reducers get called, using the action that was dispatched's type key on the action object]
    F --> G[your user reducer is invoked and matches to logoutUser]
    G --> H[your logout reducer sets the global application state to user.isVerified to true]
    H --> I[any component that is using the user app state like 4 or five right now]
    I --> J[all of the associated actions, reducers, and dispatches are called again]
```

all of your other state modifications are only occurring locally, these steps needs to be both followed and considered for all of the current useState calls you have. you should also consider exactly what is needed to listen to from global app state, and select the smallest possible slice, not just like the whole thing, because it causes rerenders